node/bft/src/primary.rs

```diff
@@ -295,6 +311,21 @@ impl<N: Network> Primary<N> {
         // If there is a batch being proposed already,
         // rebroadcast the batch header to the non-signers, and return early.
         if let Some(proposal) = self.proposed_batch.read().as_ref() {
+            // Ensure that the storage is caught up to the proposal before proceeding to rebroadcast this.
+            if self.current_round() < proposal.round()
+                || proposal
+                    .batch_header()
+                    .previous_certificate_ids()
+                    .iter()
+                    .any(|id| !self.storage.contains_certificate(*id))
+            {
+                // TODO (raychu86): Explicitly request the missing certificates from peers.
+                debug!(
+                    "Cannot propose a batch for round {} - the current storage is not caught up to the proposed batch.",
```

Comment:
You should print `the current storage ({current_storage})`

Reply:
Resolved with [7aaf276](https://github.com/AleoHQ/snarkOS/pull/3239/commits/7aaf2765bc7c53152f87b913c78b62892db54792) and [53eca2f](https://github.com/AleoHQ/snarkOS/pull/3239/commits/53eca2ffd697a50145a8f341bf60d6573872ac03)